lib/request_handler/body_parser.rb

```diff
@@ -25,7 +25,7 @@ def run
 
     def flattened_request_body
       body = request_body.fetch('data') do
-        raise ExternalArgumentError, body: 'must contain data'
+        raise BodyParamsError, body: 'must contain data'
```

Comment:
If the app wants to handle certain error different it's way easier to do this by means of the error class instead of the error message. Checking the error message to find out what happend can be very brittle because it means we i.e. can't simply improve the error messages without breaking something.

With this change we can do this

```ruby
begin
 ...
rescue BodyParamsError
 p "body was invalid"
rescue FilterParamsError
 p "filter params where invalid"
resuce ExternalArgumentError
 p "something else was invalid"
end
```

Also with the way it's implemented with this PR you can also still simple catch all `ExternalArgumentErrors` if you don't need to distinguish between what exactly went wrong.